sentry-android-fragment/src/main/java/io/sentry/android/fragment/FragmentLifecycleIntegration.kt

```diff
@@ -0,0 +1,66 @@
+package io.sentry.android.fragment
+
+import android.app.Activity
+import android.app.Application
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Bundle
+import androidx.fragment.app.FragmentActivity
+import io.sentry.IHub
+import io.sentry.ILogger
+import io.sentry.Integration
+import io.sentry.SentryLevel.DEBUG
+import io.sentry.SentryOptions
+import java.io.Closeable
+
+class FragmentLifecycleIntegration(
+    private val application: Application,
+    private val logger: ILogger
+) : ActivityLifecycleCallbacks, Integration, Closeable {
+
+    private lateinit var hub: IHub
+
+    override fun register(hub: IHub, options: SentryOptions) {
+        this.hub = hub
+
+        application.registerActivityLifecycleCallbacks(this)
```

Comment:
Differently from `ActivityLifecycleIntegration`, I don't add a flag in `SentryAndroidOptions` and [I don't check if this integration is enabled.](https://github.com/getsentry/sentry-java/blob/112c46d04edfcdf1507f60bac8a401854c834457/sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java#L70)

The reason for that is that it's an opt-in feature and the user has to add a new module and integration manually. I think there's no point to hide it behind a flag in this case.

Reply:
I agree, its not needed